nixos/modules/services/backup/bacula.nix

```diff
@@ -108,13 +128,95 @@ let
     ${dir_cfg.extraConfig}
     '';
 
+  tlsOptions = {...}:
+  {
+    options = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Specifies if TLS should be enabled if this is false TLS will be
+          completely disabled, even if tls.require is true.
+        '';
+      };
+      require = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = mdDoc ''
+          Require TLS or TLS-PSK encryption. This directive is ignored unless
+          one of TLS Enable or TLS PSK Enable is set to true.
+          If TLS is not required while TLS or TLS-PSK are enabled, then the
+          Bacula component will connect with other components either with or
+          without TLS or TLS-PSK
+
+          If TLS or TLS-PSK is enabled and TLS is required, then the Bacula
+          component will refuse any connection request that does not use TLS.
```

Comment:
I think this description is a bit confusing. 
TLS enable is never set to `true`, as the Bacula configuration syntax uses yes/no. `${submodulePath}.tls.enable` is set to `true`. 
TLS-PSK enable is not exposed as a NixOS option, so how is a user supposed to set that to true?